perses/dispersed/feptasks.py

```diff
@@ -291,222 +158,210 @@ def __init__(self, alchemical_functions: dict, splitting: str, temperature: unit
+        sampler_state.apply_to_context(context, ignore_velocities=True)
+        context.setVelocitiesToTemperature(thermodynamic_state.temperature) #randomize velocities @ temp
+        init_state = context.getState(getEnergy=True)
+        initial_energy = self._beta * (init_state.getPotentialEnergy() + init_state.getKineticEnergy())
+        sampler_state.update_from_context(context, ignore_velocities=True)
 
-        # Subclasses may implement _before_integration().
-        self._before_integration(context, thermodynamic_state)
+        if self._direction == 'forward':
+            assert context.getParameter('lambda_sterics_core') == 0.0, f"Direction was specified as {self._direction} but initial master lambda is defined as {context.getParameter('lambda_sterics_core')}" #lambda_sterics_core maps master_lambda (see lambda_protocol.py)
+        elif self._direction == 'reverse':
+            assert context.getParameter('lambda_sterics_core') == 1.0, f"Direction was specified as {self._direction} but initial master lambda is defined as {context.getParameter('lambda_sterics_core')}" #lambda_sterics_core maps master_lambda (see lambda_protocol.py)
 
-        self._cumulative_work[0] = integrator.get_protocol_work(dimensionless=True)
 
-        if self._cumulative_work[0] != 0.0:
-            raise RuntimeError("The initial cumulative work after reset was not zero.")
+        # reset the integrator after it is bound to a context and before annealing
+        integrator.reset()
 
-        if self._measure_shadow_work:
-            initial_energy = self._beta * (sampler_state.potential_energy + sampler_state.kinetic_energy)
-        #loop through the number of times we have to apply in order to collect the requested work and trajectory statistics.
-        for iteration in range(self._number_of_step_moves):
+        #save the initial configuration
+        if self._save_configuration:
+            if self._subset_atoms is None:
+                self._trajectory_positions.append(sampler_state.positions[:, :].value_in_unit_system(unit.md_unit_system))
+            else:
+                self._trajectory_positions.append(sampler_state.positions[self._subset_atoms, :].value_in_unit_system(unit.md_unit_system))
 
-            try:
-                integrator.step(self._work_save_interval)
-            except Exception as e:
-                self._trajectory = md.Trajectory(self._trajectory_positions, self._topology, unitcell_lengths=self._trajectory_box_lengths, unitcell_angles=self._trajectory_box_angles)
-                raise e
-            self._current_protocol_work = integrator.get_protocol_work(dimensionless=True)
-            self._protocol_work[iteration+1] = self._current_protocol_work
-            self._cumulative_work[iteration+1] = self._protocol_work [iteration+1]
+            #get the box angles and lengths
+            a, b, c, alpha, beta, gamma = mdtrajutils.unitcell.box_vectors_to_lengths_and_angles(*sampler_state.box_vectors)
```

Comment:
Is it just me or is it weird to call it `mdtrajutils`

I guess this is done because a) the `utils` namespace is used, and b) you don't want to import all of mdtraj...

I get it but it's funny to look at